lib/dialects/abstract/query-generator.js

```diff
@@ -901,6 +901,8 @@ module.exports = (function() {
             }
           }
           attr = [attr[0], self.quoteIdentifier(attr[1])].join(' AS ');
+        } else if (mainModel && mainModel._isGeometryAttribute(attr)){
+          attr = 'ST_AsGeoJSON('+self.quoteIdentifiers(attr)+') AS '+ self.quoteIdentifiers(attr);
```

Comment:
attr may be composed of both the column name and the alias, comma separated. Then the lookup doesn't work.